src/main/java/org/spongepowered/api/command/Command.java

```diff
@@ -0,0 +1,312 @@
+         * {@link #extendedDescription(Text)}</p>
+         *
+         * @param description The description to use, or {@code null} for no
+         *                    description
+         * @return This builder, for chaining
+         */
+        Builder description(@Nullable Text description);
+
+        /**
+         * Provides the logic of the command.
+         *
+         * <p>This is only optional if child commands are specified.</p>
+         *
+         * @param executor The {@link CommandExecutor} that will run the command
+         * @return This builder, for chaining
+         */
+        Builder executor(CommandExecutor executor);
+
+        /**
+         * Provides the description for this command.
+         *
+         * <p>A one line summary should be provided to
+         * {@link #description(Text)}</p>
+         *
+         * @param extendedDescription The description to use, or {@code null}
+         *                            for no description.
+         * @return This builder, for chaining
+         */
+        Builder extendedDescription(@Nullable Text extendedDescription);
+
+        /**
+         * The flags that this command should accept. See {@link Flags}.
+         *
+         * @param flags The {@link Flags} to accept
+         * @return This builder, for chaining
+         */
+        Builder flags(Flags flags);
+
+        /**
+         * Determines how an argument string should be split.
+         *
+         * <p>Defaults to splitting on spaces, ignoring spaces in quoted
+         * regions.</p>
+         *
+         * @param tokenizer The {@link InputTokenizer} to use
+         * @return This builder, for chaining
+         */
+        Builder inputTokenizer(InputTokenizer tokenizer);
+
+        /**
+         * The parameter set to use when parsing arguments. Parameters will be
+         * used in the order provided here.
+         *
+         * @param parameters The {@link Parameter}s to use
+         * @return This builder, for chaining
+         */
+        Builder parameters(Parameter... parameters);
+
+        /**
+         * The parameter set to use when parsing arguments. Parameters will be
+         * used in the order provided here.
+         *
+         * @param parameters The {@link Parameter}s to use
+         * @return This builder, for chaining
+         */
+        Builder parameters(Iterable<Parameter> parameters);
+
+        /**
+         * The permission that a {@link CommandSource} requires to run this
+         * command, or {@code null} if no permission is required.
+         *
+         * @param permission The permission that is required, or {@code null}
+         *                   for no permission
+         * @return This builder, for chaining
+         */
+        Builder permission(@Nullable String permission);
+
+        /**
+         * Builds this command.
+         *
+         * @return The command, ready for registration
+         */
+        Command build();
```

Comment:
Nothing Sponge-specific, just seemed suspicious given what could possibly go wrong in the build.